File: hackerrank/easy/0018-cats-and-a-mouse/python/solution.py
## solution.py

def cat_and_mouse(x: int, y: int, z: int) -> str:
    da = abs(z - x)
    db = abs(z - y)
    if da == db:
        return "Mouse C"
    return "Cat A" if da < db else "Cat B"


# --- demo run ---
if __name__ == "__main__":
    print(cat_and_mouse(1, 2, 3))  # Cat B
    print(cat_and_mouse(1, 3, 2))  # Mouse C

File: hackerrank/easy/0018-cats-and-a-mouse/js/solution.mjs
// solution.mjs

export function catAndMouse(x, y, z) {
  const da = Math.abs(z - x);
  const db = Math.abs(z - y);
  if (da === db) return "Mouse C";
  return da < db ? "Cat A" : "Cat B";
}

// --- demo run ---
console.log(catAndMouse(1, 2, 3)); // Cat B
console.log(catAndMouse(1, 3, 2)); // Mouse C